pkg/controller/wip/controller.go

```diff
@@ -552,35 +676,102 @@ func (c *controller) reconcileBinding(binding *v1alpha1.Binding) {
 	glog.V(4).Infof("Creating client for Broker %v, URL: %v", broker.Name, broker.Spec.URL)
 	brokerClient := c.brokerClientCreateFunc(broker.Name, broker.Spec.URL, username, password)
 
-	// TODO: parameters
+	if binding.DeletionTimestamp == nil { // Add or update
+		glog.V(4).Infof("Adding/Updating Binding %v/%v", binding.Namespace, binding.Name)
+
+		servicePlan := findServicePlan(instance.Spec.PlanName, serviceClass.Plans)
+		if servicePlan == nil {
+			glog.Errorf("Instance %v/%v references a non-existent ServicePlan %v on ServiceClass %v", instance.Namespace, instance.Name, servicePlan.Name, serviceClass.Name)
+			c.updateBindingCondition(
+				binding,
+				v1alpha1.BindingConditionReady,
+				v1alpha1.ConditionFalse,
+				"ReferencesNonexistentServicePlan",
+				"The Binding references an Instance which references ServicePlan that does not exist",
+			)
+			return
+		}
+
+		// TODO: parameters
+		request := &brokerapi.BindingRequest{
+			ServiceID: serviceClass.OSBGUID,
+			PlanID:    servicePlan.OSBGUID,
+		}
+		response, err := brokerClient.CreateServiceBinding(instance.Spec.OSBGUID, binding.Spec.OSBGUID, request)
+		if err != nil {
+			glog.Errorf("Error creating Binding %v/%v for Instance %v/%v of ServiceClass %v at Broker %v: %v", binding.Name, binding.Namespace, instance.Namespace, instance.Name, serviceClass.Name, broker.Name, err)
+			c.updateBindingCondition(
+				binding,
+				v1alpha1.BindingConditionReady,
+				v1alpha1.ConditionFalse,
+				"BindCallFailed",
+				"Bind call failed")
+			return
+		}
+		err = c.injectBinding(binding, &response.Credentials)
+		if err != nil {
+			glog.Errorf("Error injecting binding results for Binding %v/%v: %v", binding.Namespace, binding.Name, err)
+			c.updateBindingCondition(
+				binding,
+				v1alpha1.BindingConditionReady,
+				v1alpha1.ConditionFalse,
+				"ErrorInjectingBindResult",
+				"Error injecting bind result",
+			)
+			return
+		}
+		c.updateBindingCondition(
+			binding,
+			v1alpha1.BindingConditionReady,
+			v1alpha1.ConditionTrue,
+			"InjectedBindResult",
+			"Injected bind result",
+		)
 
-	request := &brokerapi.BindingRequest{
-		ServiceID: serviceClass.OSBGUID,
-		PlanID:    servicePlan.OSBGUID,
+		glog.V(5).Infof("Successfully bound to Instance %v/%v of ServiceClass %v at Broker %v", instance.Namespace, instance.Name, serviceClass.Name, broker.Name)
+
+		return
 	}
 
-	response, err := brokerClient.CreateServiceBinding(instance.Spec.OSBGUID, binding.Spec.OSBGUID, request)
+	// Soft delete...
+	if len(binding.Finalizers) > 0 && binding.Finalizers[0] == "BindingFinalizer" {
+		glog.V(4).Infof("Deleting Binding %v/%v", binding.Namespace, binding.Name)
```

Comment:
In this particular location, I think we're talking about the fate of the service catalog object. I'm going to change this to read "Finalizing Binding...", which is more accurate. Same goes for other types.